### PR TITLE
Add async/await/sync/yield and toImport/ExportBuilder

### PR DESCRIPTION
**Pull request by matanlurey**
## 1.0.0-alpha+8

- Add support for `async`, `sync`, `sync*` functions
- Add support for ann expression to `asAwait`, `asYield`, `asYieldStar`
- Add `toExportBuilder` and `toImportBuilder` to types and references

Closes https://github.com/dart-lang/code_builder/issues/40
Closes https://github.com/dart-lang/code_builder/issues/42

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 1.0.0-alpha+8
```
Comment (yjbanov) [inline CHANGELOG.md]: `+9` :)

File: lib/src/builders/method.dart [hunk 2]
```diff
@@ -18,33 +18,57 @@ import 'package:code_builder/src/builders/type.dart';
...
-    [Iterable<ValidConstructorMember> members = const []]) {
+ConstructorBuilder constructorNamed(
+  String name, [
+  Iterable<ValidConstructorMember> members = const [],
+]) {
   return _constructorImpl(name: name, members: members);
 }
 
+/// Various types of modifiers for methods.
+class MethodModifier implements ValidMethodMember {
```
Comment (yjbanov) [inline lib/src/builders/method.dart]: Are things like `static` and `final` considered modifiers? I'm also wondering if `package:meta` should be supported out-of-the-box (as a separate feature request probably), for things like `@protected` and `@mustCallSuper`.

File: lib/src/builders/method.dart [hunk 3]
```diff
@@ -176,6 +212,7 @@ abstract class MethodBuilder
   /// Creates a new [MethodBuilder].
   factory MethodBuilder(
     String name, {
+    MethodModifier modifier,
```
Comment (yjbanov) [inline lib/src/builders/method.dart]: Can a method have multiple modifiers? I'm not terribly versed in the Dart terminology, but in Java `private`, `protected`, `static`, `final`, etc are [all considered "modifiers"](https://www.tutorialspoint.com/java/java_modifier_types.htm) and can be combined.

File: pubspec.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name: code_builder
-version: 1.0.0-alpha+7
+version: 1.0.0-alpha+8
```
Comment (yjbanov) [inline pubspec.yaml]: `+9`

File: test/builders/method_test.dart [hunk 5]
```diff
@@ -170,6 +170,33 @@ void main() {
...
+  test('should emit an async method', () {
+    expect(
+      method('fetch', [MethodModifier.async]),
+      equalsSource(r'''
+        fetch() async {}
+      '''),
+    );
+  });
+
+  test('should emit an async* method', () {
```
Comment (yjbanov) [inline test/builders/method_test.dart]: `async*` is not complete without a `await for/in` :)

File: lib/src/builders/method.dart (hunk 2 above)
Comment (matanlurey) [inline lib/src/builders/method.dart]: I don't believe so (I am just copying analyzer terminology though)

File: test/builders/method_test.dart (hunk 5 above)
Comment (matanlurey) [inline test/builders/method_test.dart]: For loops/while loops in a future PR :)

File: lib/src/builders/method.dart (hunk 3 above)
Comment (zoechi) [inline lib/src/builders/method.dart]: `static final` or `static const` are valid

**Pull request merged**